Gifski/util.swift

```diff
@@ -2136,3 +2145,20 @@ extension NSView {
 		return window?.convertToScreen(convert(bounds, to: nil))
 	}
 }
+
+extension Collection {
+	/// Returns the element at the specified index if it is within bounds, otherwise nil.
+	subscript(safe index: Index) -> Element? {
+		return indices.contains(index) ? self[index] : nil
+	}
+}
+
+protocol Copyable {
+	init(instance: Self)
+}
+
+extension Copyable {
+	func copy() -> Self {
+		return Self.init(instance: self)
```

Comment:
```suggestion
		return Self(instance: self)
```

Reply:
oh interesting, it works without specifying `.init` - I remember some time ago you wouldn't be able to do so in protocol extension, wonder when that changed